rednose.py

```diff
@@ -33,367 +33,371 @@
+            termstyle.green(
+                " (%s test%s passed)" % (self.success, self._plural(self.success))
+            )
+        )
+        self._outln()
+
+    def _plural(self, num):
+        return '' if num == 1 else 's'
+
+    def _line(self, color=termstyle.reset, char='-'):
+        """
+        Print a line of separator characters (default '-') in the given colour (default black).
+        """
+        self._outln(color(char * line_length))
+
+    def getShortDescription(self, test):
+        doc_first_line = test.test.shortDescription()
+        if self.descriptions and doc_first_line:
+            return self.indent + doc_first_line
+        return self.indent + test.test._testMethodName
+
+    def getLongDescription(self, test):
+        doc_first_line = test.test.shortDescription()
+        if self.descriptions and doc_first_line:
+            return '\n'.join((str(test), doc_first_line))
+        return str(test)
+
+    def getClassDescription(self, test):
+        test_class = test.test.__class__
+        doc = test_class.__doc__
+        if self.descriptions and doc:
+            return doc.split('\n')[0].strip()
+        return strclass(test_class)
+
+    def _print_test(self, type_, color):
+        self.total += 1
+        if self.verbose:
+            self._outln(color(type_))
+        else:
+            if type_ == failure:
+                short_ = 'F'
+            elif type_ == error:
+                short_ = 'E'
+            elif type_ == skip:
+                short_ = '-'
+            elif type_ == expected_failure:
+                short_ = "X"
+            elif type_ == unexpected_success:
+                short_ = "U"
+            else:
+                short_ = '.'
+
+            self._out(color(short_))
+            if self.total % line_length == 0:
+                self._outln()
+
+    def _out(self, msg='', newline=False):
+        self.stream.write(msg)
+        if newline:
+            self.stream.write('\n')
+
+    def _outln(self, msg=''):
+        self._out(msg, True)
+
+    def _add_report(self, report):
+        failure_type, test, err = report
+        self.reports.append(report)
+        if self.immediate:
+            self._outln()
+        self._report_test(len(self.reports), *report)
+
+    def addFailure(self, test, err):
+        self.failure += 1
+        self._print_test(failure, termstyle.red)
+        self._add_report((failure, test, err))
+
+    def addError(self, test, err):
+        if err[0].__name__ == 'SkipTest':
+            self.addSkip(test, err)
+            return
+        self.error += 1
+        self._print_test(error, termstyle.yellow)
+        self._add_report((error, test, err))
+
+    def addSuccess(self, test):
+        self.success += 1
+        self._print_test(success, termstyle.green)
+
+    def addSkip(self, test=None, err=None):
+        self.skip += 1
+        self._print_test(skip, termstyle.blue)
+
+    def addExpectedFailure(self, test, err):
```

Comment:
i don't think nose uses these two? but when I was reading the source code fro unittest I saw them. Also, forgot to set the variables in the constructor.